app/src/test/java/org/oppia/android/app/parser/StringToFractionParserTest.kt

```diff
@@ -0,0 +1,296 @@
+import org.oppia.android.util.accessibility.AccessibilityTestModule
+import org.oppia.android.util.caching.testing.CachingTestModule
+import org.oppia.android.util.gcsresource.GcsResourceModule
+import org.oppia.android.util.logging.LoggerModule
+import org.oppia.android.util.logging.firebase.FirebaseLogUploaderModule
+import org.oppia.android.util.parser.GlideImageLoaderModule
+import org.oppia.android.util.parser.HtmlParserEntityTypeModule
+import org.oppia.android.util.parser.ImageParsingModule
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Tests for [StringToFractionParser]. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(application = StringToFractionParserTest.TestApplication::class, qualifiers = "port-xxhdpi")
+class StringToFractionParserTest {
+
+  @Inject
+  lateinit var context: Context
+
+  private lateinit var stringToFractionParser: StringToFractionParser
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+    stringToFractionParser = StringToFractionParser()
+  }
+
+  @Test
+  fun testParser_submitTimeError_returnsValid() {
+    val error = stringToFractionParser.getSubmitTimeError("1/2")
+      .getErrorMessageFromStringRes(context)
+    assertThat(error).isEqualTo(null)
+  }
+
+  @Test
+  fun testParser_submitTimeError_extraSpaces_returnsValid() {
+    val error = stringToFractionParser.getSubmitTimeError(" 1   / 2 ")
+      .getErrorMessageFromStringRes(context)
+    assertThat(error).isEqualTo(null)
+  }
+
+  @Test
+  fun testParser_submitTimeError_tooLong_returnsNumberTooLong() {
+    val error = stringToFractionParser.getSubmitTimeError("0123456789")
+      .getErrorMessageFromStringRes(context)
+    assertThat(error)
+      .isEqualTo("None of the numbers in the fraction should have more than 7 digits.")
+  }
+
+  @Test
+  fun testParser_submitTimeError_atLengthLimit_returnsValid() {
+    val error = stringToFractionParser.getSubmitTimeError("1234567/1234567")
+      .getErrorMessageFromStringRes(context)
+    assertThat(error).isEqualTo(null)
+  }
+
+  @Test
+  fun testParser_submitTimeError_returnsDivisionByZero() {
+    val error = stringToFractionParser.getSubmitTimeError("123/0")
+      .getErrorMessageFromStringRes(context)
+    assertThat(error).isEqualTo("Please do not put 0 in the denominator")
+  }
+
+  @Test
+  fun testParser_submitTimeError_ambiguousSpacing_returnsInvalidFormat() {
+    val error = stringToFractionParser.getSubmitTimeError("1 2 3/4")
+      .getErrorMessageFromStringRes(context)
+    assertThat(error).isEqualTo("Please enter a valid fraction (e.g., 5/3 or 1 2/3)")
+  }
+
+  @Test
+  fun testParser_submitTimeError_emptyString_returnsInvalidFormat() {
+    val error = stringToFractionParser.getSubmitTimeError("")
+      .getErrorMessageFromStringRes(context)
+    assertThat(error).isEqualTo("Please enter a valid fraction (e.g., 5/3 or 1 2/3)")
```

Comment:
Here & elsewhere: I really like that you're leveraging ``getErrorMessageFromStringRes`` for clearer tests. However, we also need to ensure the returned enums are correct since that's part of the contract. I suggest updating the tests to verify the correct enum constant is returned, and have other tests to verify that each enum constant provides its correct error message.

Reply:
Done, added enum testing.